src/Features/CSharp/Portable/DocumentationComments/DocumentationCommentSnippetService.cs

```diff
@@ -111,22 +124,108 @@ protected override List<string> GetDocumentationCommentStubLines(MemberDeclarati
+            {
+                var expression = throwExpressionOrStatement switch
+                {
+                    ThrowExpressionSyntax throwExpression => throwExpression.Expression,
+                    ThrowStatementSyntax throwStatement => throwStatement.Expression,
+                    _ => throw ExceptionUtilities.Unreachable
+                };
+
+                if (expression.IsKind(SyntaxKind.NullLiteralExpression))
+                {
+                    // `throw null;` throws NullReferenceException at runtime.
+                    var exception = hasUsingSystem ? nameof(NullReferenceException) : $"{nameof(System)}.{nameof(NullReferenceException)}";
+                    if (seenExceptionTypes.Add(exception))
+                    {
+                        yield return exception;
+                    }
+                }
+                else if (expression is not null)
+                {
+                    var type = model.GetTypeInfo(expression, cancellationToken).Type;
+                    if (type is not null && !IsExceptionCaughtAndNotRethrown(throwExpressionOrStatement, type, model, cancellationToken))
+                    {
+                        var exception = type.ToDisplayString();
+                        if (seenExceptionTypes.Add(exception))
+                        {
+                            if (namespacesInScope.Contains(n => n.ToDisplayString() == type.ContainingNamespace.ToString()))
+                            {
+                                yield return type.ToDisplayString(s_format).Replace('<', '{').Replace('>', '}');
+                            }
+                            else
+                            {
+                                yield return exception.Replace('<', '{').Replace('>', '}');
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private static bool IsExceptionCaughtAndNotRethrown(SyntaxNode throwExpressionOrStatement, ITypeSymbol exceptionType, SemanticModel model, CancellationToken cancellationToken)
+        {
+            var tryStatement = throwExpressionOrStatement.FirstAncestorOrSelfUntil<TryStatementSyntax>(n => n is CatchClauseSyntax or LocalFunctionStatementSyntax or MemberDeclarationSyntax);
+            if (tryStatement is null || tryStatement.Catches.Count == 0)
+            {
+                return false;
+            }
+
+            foreach (var catchClause in tryStatement.Catches)
+            {
+                if (catchClause.Filter is not null)
+                {
+                    // It's possible that exception isn't caught.
+                    continue;
+                }
+
+                if (model.GetOperation(catchClause, cancellationToken) is ICatchClauseOperation catchClauseOperation)
+                {
+                    if (exceptionType.InheritsFromOrEquals(catchClauseOperation.ExceptionType))
```

Comment:
i see.  you use semantics for this.  i'm ok with this doing a simple thing and just comparing the last portion of the name for equality.  it will be more than good enough and will avoid semantics.